src/app_conn_pool.rs

```diff
@@ -0,0 +1,334 @@
+use crate::config::{AllowedFns, Configuration};
+use crate::{HcHttpGatewayError, HcHttpGatewayResult};
+use futures::future::BoxFuture;
+use holochain_client::{
+    AdminWebsocket, AppWebsocket, AuthorizeSigningCredentialsPayload, CellInfo, ClientAgentSigner,
+    ConductorApiError, ConnectRequest, GrantedFunctions, IssueAppAuthenticationTokenPayload,
+    Timestamp, WebsocketConfig,
+};
+use holochain_types::app::InstalledAppId;
+use holochain_types::websocket::AllowedOrigins;
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::sync::{Arc, RwLock};
```

Comment:
Could we use `tokio::sync::RwLock` for consistency?

Reply:
I don't think consistency is the goal here. I'd prefer to use the stdlib but the `app_clients` lock has to be held across `.await` points. Is there a benefit to using a tokio RwLock instead of the stdlib one?

Reply:
I asked because I noticed that the `cached_app_port` is under the std lib `RwLock` while the rest are on tokio's, checking if this was intentional.